File: exercises/ddd_infrastructure_layer/java/src/test/java/helpers/ApplicationFixtures.java
package helpers;

import java.time.ZoneId;
import java.time.ZonedDateTime;
import java.util.UUID;

import domain.application.Application;
import domain.application.ApplicationId;
import domain.application.ApplicationReceivedDateTime;
import domain.application.EmailAddress;
import domain.application.FullName;
import domain.application.MotivationLetter;
import domain.application.PhoneNumber;

public class ApplicationFixtures {
    public static final ZonedDateTime FIXED_DATE_TIME = ZonedDateTime.of(2023, 1, 1, 12, 0, 0, 0, ZoneId.of("UTC"));

    public static Application buildEmptyApplication(UUID id) {
        ApplicationId appId = new ApplicationId(id);
        ApplicationReceivedDateTime receivedDateTime = new ApplicationReceivedDateTime(FIXED_DATE_TIME);
        Application app = new Application(appId, receivedDateTime);
        return app;
    }

    public static Application buildApplicationWithBasicDetails(UUID id) {
        Application app = buildEmptyApplication(id);
        FullName fullName = new FullName("Alex Example");
        PhoneNumber phoneNumber = new PhoneNumber("447700900123");
        EmailAddress emailAddress = new EmailAddress("alex@example.com");
        app.fillOutBasicDetails(fullName, emailAddress, phoneNumber);
        return app;
    }

    public static Application buildApplicationWithMotivationLetter(UUID id) {
        Application app = buildApplicationWithBasicDetails(id);
        app.fillOutMotivationLetter(new MotivationLetter("I am very motivated."));
        return app;
    }
}

File: exercises/ddd_infrastructure_layer/typescript/test/helpers/applicationFixtures.ts
import { Application } from "../../src/domain/application/Application";
import { ApplicationId } from "../../src/domain/application/ApplicationId";
import { ApplicationReceivedDateTime } from "../../src/domain/application/ApplicationReceivedDateTime";
import { EmailAddress } from "../../src/domain/application/EmailAddress";
import { FullName } from "../../src/domain/application/FullName";
import { MotivationLetter } from "../../src/domain/application/MotivationLetter";
import { PhoneNumber } from "../../src/domain/application/PhoneNumber";

export const fixedDateTime = "2024-06-01T12:00:00Z";

export function buildEmptyApplication(id: string): Application {
  const receivedDateTime = new ApplicationReceivedDateTime(fixedDateTime);
  const app = new Application(new ApplicationId(id), receivedDateTime);
  return app;
}

export function buildApplicationWithBasicDetails(id: string): Application {
  const app = buildEmptyApplication(id);
  const fullName = new FullName("Alex Example");
  const phoneNumber = new PhoneNumber("447700900123");
  const emailAddress = new EmailAddress("alex@example.com");
  app.fillOutBasicDetails(fullName, emailAddress, phoneNumber);
  return app;
}

export function buildApplicationWithMotivationLetter(id: string): Application {
  const app = buildApplicationWithBasicDetails(id);
  const motivationLetter = new MotivationLetter("I love building things");
  app.submitMotivationLetter(motivationLetter);
  return app;
}
